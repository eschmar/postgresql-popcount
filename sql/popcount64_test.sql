CREATE EXTENSION popcount;

SELECT popcount64(15::bit(4));
SELECT popcount64(7::bit(4));

SELECT popcount64(15::bit(8));
SELECT popcount64(7::bit(8));

SELECT popcount64(15::bit(12));
SELECT popcount64(7::bit(12));

SELECT popcount64(15::bit(32));
SELECT popcount64(7::bit(32));

SELECT popcount64(30::bit(32));
SELECT popcount64(127::bit(32));

SELECT sum(popcount64(x::bit(512))) from generate_series(1,1000000) x;
SELECT sum(popcount64(x::bit(500))) from generate_series(1,1000000) x;
