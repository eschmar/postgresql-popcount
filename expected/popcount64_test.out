CREATE EXTENSION popcount;
ERROR:  extension "popcount" already exists
SELECT popcount64(15::bit(4));
 popcount64 
------------
          4
(1 row)

SELECT popcount64(7::bit(4));
 popcount64 
------------
          3
(1 row)

SELECT popcount64(15::bit(8));
 popcount64 
------------
          4
(1 row)

SELECT popcount64(7::bit(8));
 popcount64 
------------
          3
(1 row)

SELECT popcount64(15::bit(12));
 popcount64 
------------
          4
(1 row)

SELECT popcount64(7::bit(12));
 popcount64 
------------
          3
(1 row)

SELECT popcount64(15::bit(32));
 popcount64 
------------
          4
(1 row)

SELECT popcount64(7::bit(32));
 popcount64 
------------
          3
(1 row)

SELECT popcount64(30::bit(32));
 popcount64 
------------
          4
(1 row)

SELECT popcount64(127::bit(32));
 popcount64 
------------
          7
(1 row)

SELECT popcount64(b'10101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010');
 popcount64 
------------
        263
(1 row)

SELECT sum(popcount64(x::bit(512))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(popcount64(x::bit(500))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

